webapp/src/components/App/Singlestudy/explore/Modelization/Map/Areas/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { useCallback, useEffect, useState } from "react";
+import {
+  NodeProperties,
+  UpdateAreaUi,
+} from "../../../../../../../common/types";
+import PanelView from "./PanelView";
+import PropertiesView from "../../../../../../common/PropertiesView";
+import ListElement from "../../../common/ListElement";
+import { AreasContainer } from "../style";
+import useAppSelector from "../../../../../../../redux/hooks/useAppSelector";
+import {
+  getCurrentAreaId,
+  getSelectedLink,
+  getSelectedNode,
+  getSelectedNodeLinks,
+} from "../../../../../../../redux/selectors";
+import { setSelectedNode } from "../../../../../../../redux/ducks/studyDataSynthesis";
+import useAppDispatch from "../../../../../../../redux/hooks/useAppDispatch";
+
+interface Props {
+  nodeList: Array<NodeProperties>;
+  onDelete?: (id: string, target?: string) => void;
+  onArea: () => void;
+  updateUI: (id: string, value: UpdateAreaUi) => void;
+}
+
+function Areas(props: Props) {
+  const { nodeList, onDelete, onArea, updateUI } = props;
+  const dispatch = useAppDispatch();
+  const [filteredNodes, setFilteredNodes] =
+    useState<Array<NodeProperties>>(nodeList);
+  const [serchedNodes, setSerchedNodes] = useState<string>();
+  const selectedNode = useAppSelector(getSelectedNode);
+  const selectedLink = useAppSelector(getSelectedLink);
+  const selectedNodeLinks = useAppSelector(getSelectedNodeLinks);
+  const currentNode = useAppSelector(getCurrentAreaId);
+
+  const filterNodes = useCallback(
```

Comment:
Move the function inside the useEffect (without search value in arg) and update its deps